### DISCUSSION TITLE
Request for Guidance: Regenerating common.pb.go for Hyperledger Fabric v2.2

### DISCUSSION THREAD

**Discussion by r4sk01** [hyperledger/fabric-protos#186] [Q&A]
Hello, 

I am currently attempting to regenerate the `common.pb.go` file for Hyperledger Fabric v2.2 and have a few queries regarding the process.

Firstly, I'd like to know which branch of the repository I should be using for this task. According to the [documentation](https://hyperledger.github.io/fabric-protos/), it appears that even the v0.1.x branch is for v2.4 releases. Can you please clarify which branch should be used for v2.2?

Secondly, as per my understanding, the process should involve compiling the [common/common.proto](https://github.com/hyperledger/fabric-protos/blob/main/common/common.proto) file. However, I am unsure about the exact steps to take in this regard. Could you please provide a detailed guide or the specific commands that I should use to properly compile this file?

I would appreciate your assistance in helping me navigate this process. Thank you in advance for your time and support.

Best regards,

**Comment by bestbeforetoday** [comment]
There is a [release-2.2](https://github.com/hyperledger/fabric-protos/tree/release-2.2) branch, which _should_ reflect the state of the protobuf definitions for Fabric v2.2. In reality there is some effort put in to ensuring that breaking changes are not introduced into the protobufs so you can _probably_ get away with using one of the more recent branches and not break anything.

In the current (main, v0.1.x and v0.2.x branches), there are Makefile targets to generate language-specific bindings:
- Go: `make genprotos`
- Java: `make javabindings`
- Node (JavaScript / TypeScript): `make nodebindings`

If you are contributing changes to this repository then you need be make changes to the **main** branch first, and then cherry-pick back to previous branches as far back as you need to go.

If you are just trying to consume the generated language bindings, there are already Go, Node and Java packages published that you can use, as referenced in the [documentation](https://hyperledger.github.io/fabric-protos/).

If you want to generate your own bindings for use in your own projects, you don't need to use the build process implemented in this repository. The [protocol buffer documentation](https://protobuf.dev/) includes information on how language bindings are generated from protobuf definitions.
